src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -4404,12 +4404,24 @@ void CodeGen::genLeaInstruction(GenTreeAddrMode* lea)
         else
         {
 #ifdef TARGET_ARM64
-            // Handle LEA with "contained" BFIZ
-            if (index->isContained() && index->OperIs(GT_BFIZ))
+
+            if (index->isContained())
             {
-                assert(scale == 0);
-                scale = (DWORD)index->gtGetOp2()->AsIntConCommon()->IconValue();
-                index = index->gtGetOp1()->gtGetOp1();
+                if (index->OperIs(GT_BFIZ))
+                {
+                    // Handle LEA with "contained" BFIZ
+                    assert(scale == 0);
+                    scale = (DWORD)index->gtGetOp2()->AsIntConCommon()->IconValue();
+                    index = index->gtGetOp1()->gtGetOp1();
+                }
+                else if (index->OperIs(GT_CAST))
+                {
+                    index = index->AsCast()->gtGetOp1();
+                }
```

Comment:
What kind of cast gets here? `CAST<long>(int)`? But then what is doing the sign/zero-extension?

Reply:
```
N006 (  4,  6) [000018] -c--G----U-                   t18 = *  CAST      long <- uint $1c0
                                                            /--*  t19    byref  
                                                            +--*  t18    long   
N007 (  6,  8) [000020] -----------                   t20 = *  LEA(b+(i*1)+0) byref 
```

Reply:
Hmm, so it is indeed `CAST<long>(int)`. There is still the question of what is doing the extension (is it perhaps unnecessary)?